v5.0.8 native build fails with latest PX4-GPSDrivers (GPSDriverUBX ctor signature changed)

### Have you checked the latest release for fixes?

- [x] Yes, I’ve tested on the latest release

### Description

**Description**

Building QGroundControl v5.0.8 natively on Linux fails after recent updates in the PX4/PX4-GPSDrivers repo. The QGC source code itself is unchanged on our side; the breakage appears to come from an upstream API change in GPSDriverUBX constructor (now requires an additional Settings argument).

**Environment**

- QGroundControl: v5.0.8
- OS: Linux (native build)
- Compiler: g++ (via cmake/ninja)
- Qt: 6.8.3 (from build log)
- Build system: CMake + Ninja
- Dependency manager: CPM (px4-gpsdrivers via CPMAddPackage)

**Steps to reproduce**

- Checkout QGC v5.0.8
- Configure + build natively (cmake/ninja)
- Build fails in src/GPS/GPSProvider.cc when compiling GPSProvider.cc

**Actual result**

Build stops with:
```

/home/.../src/GPS/GPSProvider.cc:223:120: error: no matching function for call to
'GPSDriverUBX::GPSDriverUBX(GPSHelper::Interface, int (*)(GPSCallbackType, void*, int, void*), GPSProvider*, sensor_gps_s*, satellite_info_s*)'

candidate: 'GPSDriverUBX::GPSDriverUBX(Interface gpsInterface, GPSCallbackPtr callback, void *callback_user,
                                       sensor_gps_s*, satellite_info_s*, Settings)'
candidate expects 6 arguments, 5 provided

```

So QGC calls the 5-argument constructor, but px4-gpsdrivers now expects 6 arguments (added Settings).

**Expected result**

v5.0.8 should build successfully using its default dependency resolution, or the dependency should be pinned to a compatible tag/commit.

Workaround (temporary fix)

Pin px4-gpsdrivers to an older compatible commit in src/GPS/CMakeLists.txt:

```
CPMAddPackage(
    NAME px4-gpsdrivers
    GITHUB_REPOSITORY PX4/PX4-GPSDrivers
    GIT_TAG caf5158061bd10e79c9f042abb62c86bc6f3e7a7
    SOURCE_SUBDIR src
)

```

With this pin, the build succeeds.

### Platform

Linux

### Flight Stack

PX4

### System Information

- QGC Version: 5.0.8
- QGC build: Self-built from source
- Operating System: Ubuntu 24 (x86_64)
- Flight Controller: Not applicable (compilation issue)
- Autopilot Software (with version): Not applicable

### Log Files and Screenshots

_No response_

--- a/src/GPS/CMakeLists.txt
+++ b/src/GPS/CMakeLists.txt
@@ -28,7 +28,7 @@ target_include_directories(${CMAKE_PROJECT_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_
 CPMAddPackage(
     NAME px4-gpsdrivers
     GITHUB_REPOSITORY PX4/PX4-GPSDrivers
-    GIT_TAG main
+    GIT_TAG caf5158061bd10e79c9f042abb62c86bc6f3e7a7
     SOURCE_SUBDIR src
 )
 

